src/main/scala/edu/colorado/plv/cuanto/jsy/binding/Parser.scala

```diff
@@ -0,0 +1,77 @@
+package edu.colorado.plv.cuanto.jsy
+package binding
+
+import edu.colorado.plv.cuanto.jsy.common.{JsyParserLike, OpParserLike, UnitOpParser}
+
+import scala.util.parsing.input.Positional
+
+/** Parse variables and bindings.
+  *
+  * @author Bor-Yuh Evan Chang
+  */
+trait ParserLike extends OpParserLike with JsyParserLike {
+  import ParserLike._
+
+  override def stmt: Parser[Expr] = {
+    rep(concreteStmt) ^^ { stmts =>
+      stmts.foldRight(None: Option[Expr]) {
```

Comment:
I think that this statement-expression bifurcation (which allows concatenation of arbitrary expressions) is the right solution here - getting into the business of dealing with newline characters and whitespace is messier, in my opinion.